metashare/consumer_utils.py

```diff
@@ -15,9 +15,7 @@ def message_to_hash(message):
 async def get_set_message_semaphore(channel_layer, message):
     msg_hash = message_to_hash(message)
     async with channel_layer.connection(0) as connection:
-        ret = await connection.setnx(msg_hash, 1)
-        await connection.expire(msg_hash, 2)
-        return ret
+        return await connection.setnx(msg_hash, 1, expire=2)
```

Comment:
@wlonk aioredis' `setnx` method does not accept an `expire` kwarg, even though the redis SETNS command does (as I discovered somewhat painfully in MetaDeploy since our testing uses a mocked redis connection). Please use the `set` method like I pointed you to in MetaDeploy: https://github.com/SFDO-Tooling/MetaDeploy/blob/master/metadeploy/consumer_utils.py#L21

Reply:
I had read a bit of the code and wondered if that was the case, but there was an indirection that made me unsure if I was looking at the right type of Connection. I threw this together at the end of a day of airplane travel and didn't get a chance to test it out, too, so it was speculative at best! Looks like this has all been sorted out though, thanks.